### PR TITLE
new feature: add privacy-router for sf-lr-mix

### PR DESCRIPTION
**Pull request by strongman1995**
**Type of change**

- [ ] Add new papers (Please tell us why you think this paper is awesome!)
- [ ] Fix the category of an existing paper/papers (Please tell us the reasons)
- [x] Add a new tool/primitive/application with a new markdown page (Thank you! Also, please tell us more about this awesome thing!)

**Description**

> Please include a summary of the change and the relevant motivation.
Supply a new component "Privacy-Router" for enabling mutual conversion between homomorphically encrypted data and secret-shared data.

### CODE REVIEW

File: secretflow/ml/linear/interconnection/router_hooks.py [hunk 1]
```diff
@@ -50,7 +148,7 @@ def unflatten():
         return list(unflatten())
 
 
-# 华控隐私路由水平 LR 方案实现
+# 隐私路由水平 LR 方案实现(WIP)
```
Comment (usafchn) [inline secretflow/ml/linear/interconnection/router_hooks.py:151]: 华控的字样不要去掉，上次王老师说要提现华控。

==

这边可能是你的分支没有 merge main 分支最新的改动？

**Review by usafchn — DISMISSED** [review]
lgtm

File: tests/ml/linear/test_fl_lr_mix.py [hunk 2]
```diff
@@ -19,6 +19,8 @@
 from secretflow.security.aggregation import SecureAggregator
 from tests.cluster import cluster, set_self_party
 
+from secretflow.ml.linear.interconnection.router_hooks import RouterLrAggrHook
```
Comment (usafchn) [inline tests/ml/linear/test_fl_lr_mix.py:22]: 把这一行注释掉或者去掉，会导致单测挂掉

Comment (strongman1995) [reply tests/ml/linear/test_fl_lr_mix.py]: 已注释

**Comment by strongman1995** [comment]
I have read the CLA Document and I hereby sign the CLA

**Comment by strongman1995** [comment]
recheck

**Comment by strongman1995** [comment]
recheck

**Review by usafchn — APPROVED** [review]

**Pull request merged**